pkg/trafficpolicy/types.go

```diff
@@ -27,8 +27,9 @@ type TrafficTarget struct {
 	HTTPRoutes  []HTTPRoute         `json:"http_route:omitempty"`
 }
 
-// RouteWeightedClusters is a struct of an HTTPRoute and associated weighted clusters
+// RouteWeightedClusters is a struct of an HTTPRoute, associated weighted clusters and the domains
 type RouteWeightedClusters struct {
 	HTTPRoute        HTTPRoute `json:"http_route:omitempty"`
 	WeightedClusters set.Set   `json:"weighted_clusters:omitempty"`
+	Domains          set.Set   `json:"domains:omitempty"`
```

Comment:
We stopped using the term `Domains` when we are actually referring to hostnames.

```suggestion
	Hostnames          set.Set   `json:"domains:omitempty"`
```